src/value/borrowed.rs

```diff
@@ -580,8 +623,100 @@ mod borrowed_value_tests {
             StructRef::from_iter(vec![("greetings", ElementRef::from(ValueRef::String("hello")))].into_iter()).into()
         ),
     )]
-    fn owned_element_accessors(elem1: ElementRef, elem2: ElementRef) {
+    fn owned_element_accessors<'a>(elem1: ElementRef<'a>, elem2: ElementRef<'a>) {
         // assert if both the element construction creates the same element
         assert_eq!(elem1, elem2);
     }
+
+    #[rstest(
+    container, length,
+    case::struct_(
+        ElementRef::new_struct(
+            vec![
+                ("greetings", ElementRef::from(ValueRef::String("hello".into()))),
+                ("name", ElementRef::from(ValueRef::String("Ion".into())))
+            ].into_iter()
+        ),
+        2
+    ),
+    case::list(
+        ElementRef::new_list(
+            vec![
+                ElementRef::from("greetings"),
+                ElementRef::from(5),
+                ElementRef::from(true)
+            ].into_iter()
+        ),
+        3
+    ),
+        case::sexp(
+            ElementRef::new_sexp(vec![ElementRef::from(5), ElementRef::from(true)].into_iter()),
+        2
+    ),
+    )]
+    fn borrowed_container_len_test(container: ElementRef, length: usize) {
+        match container.ion_type() {
+            IonType::List | IonType::SExpression => {
+                // check length for given sequence value
+                assert_eq!(container.as_sequence().unwrap().len(), length);
+            }
+            IonType::Struct => {
+                // check length for given struct value
+                assert_eq!(container.as_struct().unwrap().len(), length);
+            }
+            _ => {
+                unreachable!("This test is only for container type elements")
+            }
+        }
+    }
+
+    #[rstest(
+    container, is_empty,
+    case::struct_(
+        ElementRef::new_struct(
+            vec![
+                ("greetings", ElementRef::from(ValueRef::String("hello".into()))),
+                ("name", ElementRef::from(ValueRef::String("Ion".into())))
```

Comment:
`ValueRef::String(&str)` wraps a `&str`, so I don't think we need the `.into()`s.